### PR TITLE
Demo: Form validation fixes

### PR DESCRIPTION
**Pull request by joe-watkins**
Integrating improvements from https://github.com/tmobile/magentaA11y/pull/114

### CODE REVIEW

File: _demos/form-validation.md [hunk 1]
```diff
@@ -9,15 +9,20 @@ title:  "Form Validation"
...
+    <li>Inline errors can appear dynamically but they should not own <code>aria-live</code> or <code>role="alert"</code>. This helps reduce screen reader announcement collision or interruptions.</li>
+    <li>Inline errors do not automatically appear while user is typing.</li>
+    <li>Upon submission, keyboard focus moves to the first field with the error.</li>
+    <li>Errors are programmatically associated with <code>aria-describedby</code> pointing to a unique ID found on the element wrapping the error message.</li>
+    <li>Required fields have the <code>required</code> attribute.</li>
+    <li>Form fields with errors have <code>aria-invalid="true"</code>.</li>
+    <li>When errors are fixed by the user, automatically remove those errors from the page and remove <code>aria-invalid="true"</code> and reference to the error in <code>aria-describedby</code>.</li>
+    <li>Form fields own <code>autocomplete</code> attributes with <a href="https://www.w3.org/TR/WCAG21/#input-purposes">appropriate values</a> which is a requirement for <a href="https://www.w3.org/WAI/WCAG22/Understanding/identify-input-purpose.html">1.3.5 Identify Input Purpose (Level AA)</a>.</li>
```
Comment (silverli) [inline _demos/form-validation.md:24]: Nice!! 🎉

File: _demos/form-validation.md [hunk 2]
```diff
@@ -9,15 +9,20 @@ title:  "Form Validation"
...
+    <li>Note: The alert popup upon successful submission is for demo purposes only and real-world confirmations should not follow this pattern.</li>
```
Comment (silverli) [inline _demos/form-validation.md:25]: perfect

**Review by silverli — APPROVED** [review]
This looks great @joe-watkins, let's ship it! Unles... did you want to add your new form example to the magenta a11y form page here: https://www.magentaa11y.com/checklist-web/form/ ??!

File: _demos/form-validation.md (hunk 2 above)
Comment (joe-watkins) [reply _demos/form-validation.md:25]: Will create a new card for this work

**Pull request merged**